lib/CL/devices/common.c

```diff
@@ -135,6 +135,39 @@ void fill_dev_image_t (dev_image_t* di, struct pocl_argument* parg,
                               &(di->elem_size));
 }
 
+/**
+ * Populates the device specific sampler data structure used by kernel
+ * from given kernel sampler argument
+ */
+void fill_dev_sampler_t (dev_sampler_t *ds, struct pocl_argument *parg)
+{
+  cl_sampler_t sampler = *(cl_sampler_t *)parg->value;
+
+  *ds = 0;
+  *ds |= sampler.normalized_coords == CL_TRUE ? CLK_NORMALIZED_COORDS_TRUE :
+      CLK_NORMALIZED_COORDS_FALSE;
+
+  switch (sampler.addressing_mode) {
```

Comment:
The `CL_ADDRESS_*` constants defined in "include/CL/cl.h" don't correspond with the `CLK_ADDRESS_*` constants as defined in "include/_kernel_c.h" and currently copied in "lib/CL/devices/dev_image.h".